src/gc/gchandletableimpl.h

```diff
@@ -6,33 +6,49 @@
 #define GCHANDLETABLE_H_
 
 #include "gcinterface.h"
+#include "objecthandle.h"
 
-class GCHandleTable : public IGCHandleTable
+class GCHandleStore : public IGCHandleStore
 {
 public:
-    virtual bool Initialize();
+    GCHandleStore(HandleTableBucket *bucket) 
+        : _underlyingBucket(bucket)
+        { }
 
-    virtual void Shutdown();
+    virtual void Uproot();
 
-    virtual void* GetGlobalHandleStore();
+    virtual bool ContainsHandle(OBJECTHANDLE handle);
 
-    virtual void* CreateHandleStore(void* context);
+    virtual OBJECTHANDLE CreateHandleOfType(Object* object, int type);
 
-    virtual void* GetHandleContext(OBJECTHANDLE handle);
+    virtual OBJECTHANDLE CreateHandleOfType(Object* object, int type, int heapToAffinitizeTo);
 
-    virtual void DestroyHandleStore(void* store);
+    virtual OBJECTHANDLE CreateHandleWithExtraInfo(Object* object, int type, void* pExtraInfo);
 
-    virtual void UprootHandleStore(void* store);
+    virtual OBJECTHANDLE CreateDependentHandle(Object* primary, Object* secondary);
 
-    virtual bool ContainsHandle(void* store, OBJECTHANDLE handle);
+    virtual ~GCHandleStore();
 
-    virtual OBJECTHANDLE CreateHandleOfType(void* store, Object* object, int type);
+private:
+    HandleTableBucket* _underlyingBucket;
```

Comment:
Yes, something like this. I was thinking that the most straightforward way to do it would be to factor the initialization from `Ref_CreateHandleTableBucket` into separate function, like:

```
HandleTableBucket * Ref_CreateHandleTableBucket(void* context)
{
    HandleTableBucket * result = new (nothrow) HandleTableBucket();
    if (result == nullptr)
        return nullptr;

    if (!Ref_InitializeHandleTableBucket(result, context))
    {
        delete result;
        return nullptr;
    }

    return result;
}

bool Ref_InitializeHandleTableBucket(HandleTableBucket * bucket, void* context)
{
     // Contains the rest of the code from Ref_CreateHandleTableBucket
     // BTW: I have noticed that Ref_CreateHandleTableBucket will need work to switch over to non-throwing new operators...
}
```